concepts/pointers/introduction.md

```diff
@@ -0,0 +1,131 @@
+# Introduction
+
+Like many other languages, C++ has pointers.
+You already know `references` and pointers are similar, but think of them as a level closer to the inner workings of your computer.
+Pointers are variables that hold memory addresses.
+They are used to directly interact with memory, enabling dynamic memory allocation and efficient data manipulation in C++.
+
+If you're new to pointers, they can feel a little mysterious but once you get used to them, they're quite straight-forward.
+
+They're a crucial part of C++, so take some time to really understand them.
+The bare-bone version in this concept is also called _dumb pointer_ or _raw pointer_.
+With modern C++ there are also `smart pointers`, the basic type is not smart at all and you have to handle all the work manually.
+
+Before digging into the details, it's worth understanding the use of pointers.
+Pointers are a way to share memory with other parts of our program, which is useful for two major reasons:
+1. When you have large amounts of data, making copies to pass between functions is very inefficient.
+  By passing the memory location of where the data is stored instead, you can dramatically reduce the resource-footprint of our programs.
+2. By passing pointers between functions, you can access and modify the single copy of the data directly, meaning that any changes made by one function are immediately visible to other parts of the program when the function ends.
+
+## General Syntax
+
+A pointer declaration in C++ involves specifying the data type the pointer points to, followed by an asterisk (`*`) and the pointer's name.
+When pointers are declared, they are not automatically initialized.
+Without explicit assignment, a pointer typically holds an indeterminate value, often referred to as a "garbage address."
+While certain compilers might initialize pointers to `nullptr`, this behavior is not guaranteed across all compilers, so it's essential not to rely on it.
+It's best practice to explicitly initialize raw pointers and verify their non-null status before utilization to avoid potential issues.
+
+```cpp
+int* ptr = nullptr; // Declares a pointer and makes sure it is not invalid
+```
+
+To assign the address of a variable to a pointer, you use the address-of operator (`&`).
+Dereferencing a pointer is done using the asterisk (`*`) operator.
```

Comment:
The standard call it the "unary `*` operator". They explain that it performs an "indirection" so if you're looking for a name you could call it the "indirection operator".